implementation/src/main/java/io/smallrye/mutiny/operators/UniCacheItemOrFailure.java

```diff
@@ -0,0 +1,43 @@
+package io.smallrye.mutiny.operators;
+
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.infrastructure.Infrastructure;
+
+import java.util.function.BiPredicate;
+
+import static io.smallrye.mutiny.helpers.ParameterValidation.nonNull;
+
+/**
+ * {@link Uni}-API group for {@link UniCache} configuration
+ */
+public class UniCacheItemOrFailure<T> {
+
+    private final Uni<T> upstream;
+
+    public UniCacheItemOrFailure(Uni<T> upstream) {
+        this.upstream = nonNull(upstream, "upstream");
+    }
+
+    /**
+     * Caches the events (item or failure) of this {@link Uni} and replays it for all further
+     * {@link io.smallrye.mutiny.subscription.UniSubscriber}.
+     *
+     * @return the new {@link Uni}. Unlike regular {@link Uni}, re-subscribing to this {@link Uni} does not re-compute
+     * the outcome but replayed the cached events until the end of time.
+     */
+    public Uni<T> indefinitely() {
+        return Infrastructure.onUniCreation(new UniCache<>(upstream));
+    }
+
+    /**
+     * Caches the events (item or failure) of this {@link Uni} as long as the validator assures them validity.
+     *
+     * @param validator A {@link BiPredicate} consuming the item as well as the failure to check the cache validity.
+     * @return the new {@link Uni}. Unlike regular {@link Uni}, re-subscribing to this {@link Uni} does re-compute
+     * the outcome only after the validator {@link BiPredicate} does not succeed, otherwise the cached events are replayed.
+     */
+    public Uni<T> whilst(BiPredicate validator) {
+        return Infrastructure.onUniCreation(new UniCache<>(upstream, validator));
+    }
+
```

Comment:
Should I provide further preconfigured predicates (and which)?
Time related stuff (as mentioned at https://github.com/smallrye/smallrye-mutiny/issues/231#issuecomment-668509556) is hard with my approach since a reference time would be refer to when the predicate was created.

Reply:
Maybe we could add a `iTemOnly()` method that would only cache the item indefinitely and re-subscribe on failure. WDYT?

I will write a comment about `atMost` as PR's comment.

Reply:
Already thought about this too, but I was not quite sure if the behaviour does confuses users:
The first event from the upstream would be propagated to own subscribers, only subscribers after the first event was received would get the re-computed event from upstream, even if it is a failure.
Item would be cached not before there's no failure from upstream anymore.
I assume you did not have a "re-subscripe until there's an item" in mind?

If the "every subscriber causes a subscription to upstream until there is an item" is what we want, I would try to describe the behaviour - maybe we than can see if that could be useful in practice.

Reply:
Ok. Added the `itemOnly` UniCacheEvents function and a test.
Not quite sure about the java-doc, may you please have a look?